src/applications/appeals/995/utils/evidence.js

```diff
@@ -95,22 +95,30 @@ export const removeNonSelectedIssuesFromEvidence = data => {
  * @param {Object} router - React router
  */
 export const onFormLoaded = props => {
-  const { formData, returnUrl, router } = props;
+  let { returnUrl } = props;
+  const { formData, router } = props;
   const { locations = [] } = formData;
-  if (showScNewForm(formData) && locations.length) {
-    formData.locations = locations.map(location => {
-      if (!location.treatmentDate) {
-        const from = location.evidenceDates?.from || '';
-        const treatmentDate = (from || '').substring(0, from.lastIndexOf('-'));
-        const noDate = !treatmentDate;
-        return {
-          ...location,
-          treatmentDate,
-          noDate,
-        };
-      }
-      return location;
-    });
+  if (showScNewForm(formData)) {
+    returnUrl = checkRedirect(formData, returnUrl);
+
+    if (locations.length) {
+      formData.locations = locations.map(location => {
+        if (!location.treatmentDate) {
```

Comment:
I think this block could use a comment as to why/what it's doing. I've read it a few times now and I can see what it does, but I don't know why

Reply:
Updated!

Reply:
Ah! That really helps! Thanks! I was confused that if the `from` date was something like `mm-dd-yyyy` then it would only be using month-day.